storage/google_cloud/google_cloud_utils.py

```diff
@@ -88,5 +90,28 @@ def upload_file_to_blob(bucket_credentials_file_path, target_blob_url, f):
     log.info(f"Uploading file to blob '{target_blob_url}'...")
     storage_client = storage.Client.from_service_account_json(bucket_credentials_file_path)
     blob = _blob_at_url(storage_client, target_blob_url)
-    blob.upload_from_file(f)
-    log.info(f"Uploaded file to blob")
+
+    try:
+        blob.upload_from_file(f)
+        log.info(f"Uploaded file to blob")
+
+    except HttpError or socket.timeout as ex:
+        if ex.resp.status not in [408, 504]:
+            raise ex
+
+        log.warning(f"Failed to upload due to {ex.resp.status}")
+        chunk_sizes = [50, 25, 12.5, 6.25]
+        num_tries = 0
+        if num_tries != 5:
+            for chunk_size in chunk_sizes:
+                log.info(f"Retrying to upload file to blob '{target_blob_url}")
+                # lower the chunk size and retry uploading
+                blob.chunk_size = chunk_size * 1024 * 1024
+                blob.upload_from_file(f)
+                if ex.resp.status == 200:
```

Comment:
If this is your only exit condition then we'll retry with lower chunk sizes even where that doesn't make sense e.g. if the request fails with unauthorised. I think if we see a new error we should just throw that and build a new handler for it if it becomes a problem for us.

Reply:
Actually, how do you ever meet this condition? We should only reach this line if ex.resp.status in [408, 504] due to the if/raise on L99